src/pages/Campaign/CampaignListAndSearch.tsx

```diff
@@ -33,34 +42,68 @@ const CampaignList = styled.div`
   border-top: 1px solid ${({ theme }) => theme.border};
 
   ${({ theme }) => theme.mediaWidth.upToMedium`
-   flex: 1;
+    flex: 1;
+    overflow-x: hidden;
   `}
-  &::-webkit-scrollbar {
-    display: block;
-    width: 4px;
-  }
-
-  &::-webkit-scrollbar-thumb {
-    background: ${({ theme }) => theme.disableText};
+  .scrollbar {
+    &::-webkit-scrollbar {
+      display: block;
+      width: 4px;
+    }
+    &::-webkit-scrollbar-thumb {
+      background: ${({ theme }) => theme.disableText};
+    }
+    overflow-x: hidden !important;
   }
 `
 
-interface CampaignListAndSearchProps {
+type CampaignListAndSearchProps = {
   onSelectCampaign: (campaign: CampaignData) => void
+  loadMoreCampaign: () => void
+  hasMoreCampaign: boolean
+  onSearchCampaign: (v: string) => void
 }
 
-const CampaignListAndSearch = ({ onSelectCampaign }: CampaignListAndSearchProps) => {
+const CampaignListAndSearch = ({
+  onSelectCampaign,
+  loadMoreCampaign,
+  hasMoreCampaign,
+  onSearchCampaign,
+}: CampaignListAndSearchProps) => {
   const [searchCampaign, setSearchCampaign] = useState('')
+  const { account } = useActiveWeb3React()
   const theme = useTheme()
 
-  const { data: campaigns, selectedCampaign } = useSelector((state: AppState) => state.campaigns)
+  const { data: campaigns, selectedCampaign, lastTimeRefreshData } = useSelector((state: AppState) => state.campaigns)
 
-  const filteredCampaigns = campaigns.filter(item =>
-    item.name.toLowerCase().includes(searchCampaign.trim().toLowerCase()),
-  )
+  const debounceSearch = useDebounce(searchCampaign, INPUT_DEBOUNCE_TIME)
+  useEffect(() => {
+    onSearchCampaign(debounceSearch)
+  }, [debounceSearch, onSearchCampaign])
+
+  const scroll = useRef<HTMLDivElement | null>(null)
+  const onRefChange = useCallback((node: HTMLDivElement) => {
+    scroll.current = node
+    if (!node?.classList.contains('scrollbar')) {
+      node?.classList.add('scrollbar')
+    }
+  }, [])
+
+  useEffect(() => {
```

Comment:
What is this for?

Reply:
scroll list to top when, change search, change network